src/main/java/org/folio/rest/support/ShelvingOrderUpdate.java

```diff
@@ -0,0 +1,197 @@
+package org.folio.rest.support;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.pgclient.PgConnection;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.Tuple;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dbschema.Versioned;
+import org.folio.rest.jaxrs.model.Item;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.persist.PgUtil;
+import org.folio.rest.persist.PostgresClient;
+
+public class ShelvingOrderUpdate {
+
+  private static final Logger log = LogManager.getLogger();
+
+  private static final String SQL_EXISTS_ITEMS_TO_UPDATE = new StringBuilder()
+      .append("SELECT EXISTS (SELECT null FROM item WHERE not (jsonb ? 'effectiveShelvingOrder'))")
+      .toString();
+
+  private static final String SQL_SELECT_ITEMS = new StringBuilder()
+      .append("SELECT id AS \"id\", ")
+      .append("jsonb AS \"itemJson\" ")
+      .append("FROM item WHERE not (jsonb ? 'effectiveShelvingOrder') ")
+      .append(" FOR UPDATE ")
```

Comment:
As far as I know `for update` means that the rows will be locked, but I don't see where it is used. Do we really need it?

Reply:
Yes, that's attempts to block selected rows (i.e. so names row blocking of table) in separate transaction where then update operation for these rows will be performed. Then at completion, the transaction going to be committed and closed. I thought that blocking of rows is a kind of tradeoff for the cases where items update are going on but some of clients tries to work with data and possibly updates them. In case of blocking rows by a relatively small sizes, these clients operations of course will be in wait state until rows blocking will be released, but from other hand that will allow to keep data in consistency and possibly probability of such case isn't so high, I mean attempt to access exactly the same data belong to blocking amount of rows.